File: cpp/climbing_stairs.cpp
#include <bits/stdc++.h>

#define ll long long int

using namespace std;

int climbing_stairs(int n)
{
    ll prev = 0, curr = 1, term = 0;

    for (size_t i = 2; i <= n; i++)
    {
        term = prev + curr;
        prev = curr;
        curr = term;
    }

    return curr % 10000007;
}

int main(void)
{
    int n = 100;
    cout << climbing_stairs(n);
    return 0;
}

File: python/climbing_stairs.py
def climbing_stairs(n: int) -> int:
    prev = 0
    curr = 1

    for _ in range(2, n + 1):
        term = prev + curr
        prev = curr
        curr = term

    return term


if __name__ == "__main__":
    n = 20
    print(climbing_stairs(n))
